provisioning/device/samples/solutions/BestPracticeSampleX509/GenerateGroupEnrollment.ps1

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+param(
+    # Intermediate certificate file name including the path
+    [Parameter(Mandatory=$true)]
+    [string] $intermediateCertName,
+
+    # Resource group name of the IoT dps.
+    [Parameter(Mandatory=$true)]
+    [string] $resourceGroup,
+
+    # IoT dps name used for running the sample.
+    [Parameter(Mandatory=$true)]
+    [string] $dpsName,
+
+    # Deletes and recreate enrollmentGroup if it already exists
+    [Parameter(Mandatory=$false)]
+    [switch] $force = $false
+)
+# GroupEnrollment Id created to use in sample.
+$groupEnrollmentId = "x509GroupEnrollment"
+
+# Check certificate file extension.
+if (($intermediateCertName.EndsWith('.pem') -ne $true) -and ($intermediateCertName.EndsWith('.cer') -ne $true)){
+    Write-Host "Certificate file type must be either '.pem' or '.cer'"
+    exit
+}
+
+# Check if the resource group exists. If not, exit.
+$resourceGroupExists = az group exists -n $resourceGroup
+if ($resourceGroupExists -ne $true){
+    Write-Host "Resource Group '$resourceGroup' does not exist. Exiting..."
+    exit
+}
+
+# Check if the dps instance exists. If not, exit.
+$dpsExists = az iot dps show --name $dpsName -g $resourceGroup 2>nul
+if ($dpsExists -eq $null){
+    Write-Host "Dps '$dpsName' does not exist under '$resourceGroup'. Exiting..."
+    exit
+}
+
+# Check if the enrollment group already exists in dps instance. If it does, delete and regenerate the group enrollment.
+Write-Host "`Checking if '$groupEnrollmentId' enrollment group already exists in '$dpsName'..."
+$groupEnrollmentExists = az iot dps enrollment-group show --dps-name $dpsName -g $resourceGroup --enrollment-id $groupEnrollmentId 2>nul
+if ($groupEnrollmentExists)
+{
```

Comment:
I prefer braces on a new line like this one, but for consistency you should choose one style and stick with it.